canonical-web-and-design/maas-ui
# feat(ui): add TLS certificate to API authentication page

## Done

- Changed name of API keys page to API authentication
- Add TLS certificate info to API authentication page

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Click on the currently logged in user's name to go to user preferences
- Check that "API keys" tab has been renamed to "API authentication"
- Go to the API authentication page and check that there is a section for TLS certificate, explaining that TLS is disabled.
- To test the "TLS enabled" option you can either:
  - Enable TLS for real using `sudo maas config-tls enable $path-to-key $path-to-cert` (if you use the snap the cert files will need to be located in `/var/snap/maas`), or;
  - Fake a TLS certificate.
    - Open `ui/src/app/preferences/views/APIAuthentication/TLSCertificate/TLSCertificate.tsx`
    - On line 11 add `import { tlsCertificate as tlsCertificateFactory } from "testing/factories";`
    - Change line 22 to `const tlsCertificate = tlsCertificateFactory();`
  - Check that the API authentication section shows the TLS certificate data, and can be downloaded 

## Fixes

Fixes canonical-web-and-design/app-tribe#822

## Screenshot

![Screenshot 2022-04-19 at 14-46-56 API authentication bolla MAAS](https://user-images.githubusercontent.com/25733845/163938866-69b12d43-78b1-4a1b-8d06-53e794fb05a1.png)
